synapse/storage/databases/main/search.py

```diff
@@ -506,16 +507,18 @@ async def search_msgs(
         # entire table from the database.
         sql += " ORDER BY rank DESC LIMIT 500"
 
-        results = await self.db_pool.execute(
-            "search_msgs", self.db_pool.cursor_to_dict, sql, *args
+        # List of tuples of (rank, room_id, event_id).
+        results = cast(
+            List[Tuple[int, str, str]],
```

Comment:
Are you sure the first arg is an int? `ts_rank_cd` (used in the postgres query) returns a float according to the examples in https://www.postgresql.org/docs/current/textsearch-dictionaries.html#TEXTSEARCH-STOPWORDS

Reply:
Oh good call, not sure why I assumed it was an `int`.

SQLite uses `rank(...)` which is an `int`: https://www.sqlite.org/windowfunctions.html#built_in_window_functions